src/main/java/ua/com/javarush/oleksandr/reddit/redditcloneabstract/controller/PostController.java

```diff
@@ -30,55 +28,53 @@ public class PostController {
 
     private final PostService postService;
     private final PostMapper postMapper;
+    private final PostRequestValidator postRequestValidator;
+    private final BindingResultService bindingResultService;
 
     @PostMapping
-    public ResponseEntity<?> createPost(@RequestBody @Valid PostRequestDTO postRequestDTO,
+    public ResponseEntity<?> createPost(@RequestBody @Valid PostRequest postRequest,
                                         BindingResult bindingResult) {
 
-        if (bindingResult.hasErrors()) {
-            return handleErrorsFromBindingResult(bindingResult);
-        }
+        postRequestValidator.validate(postRequest, bindingResult);
+        bindingResultService.handle(bindingResult, PostCreationException::new);
 
-        var post = postMapper.dtoToPost(postRequestDTO);
+        var post = postMapper.dtoToPost(postRequest);
         postService.save(post);
 
         URI location = buildPostLocationUri(post.getId());
 
-        return ResponseEntity.created(location).build();
+        return ResponseEntity.status(HttpStatus.CREATED).body(location);
```

Comment:
Why should we choose to include the URL of a newly created resource in the response body rather than using the Location header, which is the standard practice in RESTful APIs?